api/src/models/Course.ts

```
@@ -316,7 +316,13 @@ courseSchema.methods.forDashboard = async function (user: IUser): Promise<ICours
     name, active, description, enrollType
   } = this;
 
-  const image = (this.image) ? (await Picture.findById(this.image)).toObject() : null;
+  let image;
+
+  try {
+    image = (await Picture.findById(this.image)).toObject();
```

```ts
const picture = await Picture.findById(this.image);
const image = picture && picture.toObject();
```
Wouldn't it be better / "more correct" to write it like this or something similar?
_(Because right now the `try`/`catch` would also absorb any potential actual errors, other than just a missing `this.image`.)_

---

Also I guess the `ICourseDashboard` should then probably make `image: IPicture;` optional (`image?: IPicture;`).

---

I need to catch various cases here. `this.image` is null and `Picture.findById(this.image)` is null. Looked more complicated to me to write this as if-else. Our goal should be that "something is wrong with the course image" does not stop the course view from working.

---

Maybe I'm mistaken, but I thought `await Picture.findById(this.image)` should return `undefined` when `this.image` is `null` (or `undefined`)?

---

Yes and it fails on `toObject` then :thinking: (see the issue https://github.com/geli-lms/geli/issues/1053)

---

Not with the aforementioned
```ts
const image = picture && picture.toObject();
```
part I would hope! ;P
If `await Picture.findById(this.image)` returns `undefined` then this way `image` will also be `undefined`, solving the problem without `try`/`catch`. Of course, if you don't like the code style with `&&`, we could just use an `if`-statement or whatever. To clarify, my only concern with your current solution is the `try`/`catch`, since it could theoretically also silence real errors in the future, that's all.

---

`const image = picture && picture.toObject as any;` would work. `ICourseDashboard` is a bit wrong about image. toObject is not castable to `IPicture` so we should use another typehint there.

---

Hmm `interface IPictureModel extends IPicture, IFileModel`, with `interface IPicture extends IFile`, but the `pictureSchema` only defines the two direct properties of `IPicture` (`breakpoints` & `mimeType`), maybe that's (part of) the problem.

---

Anyway, if you think that should have its own issue the `as any` solution _(or whatever replaces the `try`/`catch`)_ is fine by me.